File: back-end/aths/src/services/emailService.js
import { transporter, getFromAddress } from '../config/email.js';

/**
 * Email Service
 * Handles sending various types of emails
 */

/**
 * Send email
 * @param {Object} options - Email options
 */
const sendEmail = async (options) => {
  const mailOptions = {
    from: getFromAddress(),
    to: options.to,
    subject: options.subject,
    html: options.html,
    text: options.text,
  };

  try {
    const info = await transporter.sendMail(mailOptions);
    console.log(`📧 Email sent to ${options.to}: ${info.messageId}`);
    return info;
  } catch (error) {
    console.error(`❌ Failed to send email to ${options.to}:`, error.message);
    throw new Error('Failed to send email');
  }
};

/**
 * Send welcome email
 * @param {String} email - Recipient email
 * @param {String} fullName - User's full name
 */
export const sendWelcomeEmail = async (email, fullName) => {
  const subject = 'Welcome to R-MAN E-Commerce!';
  const html = `
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
          .container { max-width: 600px; margin: 0 auto; padding: 20px; }
          .header { background-color: #4CAF50; color: white; padding: 20px; text-align: center; }
          .content { padding: 20px; background-color: #f9f9f9; }
          .footer { padding: 20px; text-align: center; font-size: 12px; color: #666; }
          .button { display: inline-block; padding: 10px 20px; background-color: #4CAF50; color: white; text-decoration: none; border-radius: 5px; margin: 10px 0; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Welcome to R-MAN E-Commerce!</h1>
          </div>
          <div class="content">
            <h2>Hello ${fullName},</h2>
            <p>Thank you for registering with R-MAN E-Commerce. We're excited to have you on board!</p>
            <p>Your account has been successfully created and you can now enjoy all our services.</p>
            <p>If you have any questions or need assistance, please don't hesitate to contact our support team.</p>
            <p>Best regards,<br>The R-MAN E-Commerce Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
  `;

  const text = `
    Welcome to R-MAN E-Commerce!
    
    Hello ${fullName},
    
    Thank you for registering with R-MAN E-Commerce. We're excited to have you on board!
    
    Your account has been successfully created and you can now enjoy all our services.
    
    If you have any questions or need assistance, please don't hesitate to contact our support team.
    
    Best regards,
    The R-MAN E-Commerce Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
  `;

  await sendEmail({ to: email, subject, html, text });
};

/**
 * Send password reset email
 * @param {String} email - Recipient email
 * @param {String} fullName - User's full name
 * @param {String} resetToken - Password reset token
 */
export const sendPasswordResetEmail = async (email, fullName, resetToken) => {
  const resetUrl = `${process.env.PASSWORD_RESET_URL}?token=${resetToken}`;
  const subject = 'Password Reset Request - R-MAN E-Commerce';
  
  const html = `
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
          .container { max-width: 600px; margin: 0 auto; padding: 20px; }
          .header { background-color: #FF5722; color: white; padding: 20px; text-align: center; }
          .content { padding: 20px; background-color: #f9f9f9; }
          .footer { padding: 20px; text-align: center; font-size: 12px; color: #666; }
          .button { display: inline-block; padding: 12px 30px; background-color: #FF5722; color: white; text-decoration: none; border-radius: 5px; margin: 15px 0; }
          .warning { background-color: #fff3cd; border-left: 4px solid #ffc107; padding: 10px; margin: 15px 0; }
          .token-box { background-color: #e9ecef; padding: 15px; border-radius: 5px; font-family: monospace; word-break: break-all; margin: 15px 0; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Password Reset Request</h1>
          </div>
          <div class="content">
            <h2>Hello ${fullName},</h2>
            <p>We received a request to reset your password for your R-MAN E-Commerce account.</p>
            <p>Click the button below to reset your password:</p>
            <div style="text-align: center;">
              <a href="${resetUrl}" class="button">Reset Password</a>
            </div>
            <p>Or copy and paste this link into your browser:</p>
            <div class="token-box">${resetUrl}</div>
            <div class="warning">
              <strong>⚠️ Security Notice:</strong>
              <ul style="margin: 5px 0;">
                <li>This link will expire in ${process.env.PASSWORD_RESET_TOKEN_EXPIRATION || '1 hour'}</li>
                <li>If you didn't request this reset, please ignore this email</li>
                <li>Never share this link with anyone</li>
              </ul>
            </div>
            <p>Best regards,<br>The R-MAN E-Commerce Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
            <p>If you're having trouble clicking the button, copy and paste the URL into your web browser.</p>
          </div>
        </div>
      </body>
    </html>
  `;

  const text = `
    Password Reset Request - R-MAN E-Commerce
    
    Hello ${fullName},
    
    We received a request to reset your password for your R-MAN E-Commerce account.
    
    Click the link below to reset your password:
    ${resetUrl}
    
    ⚠️ Security Notice:
    - This link will expire in ${process.env.PASSWORD_RESET_TOKEN_EXPIRATION || '1 hour'}
    - If you didn't request this reset, please ignore this email
    - Never share this link with anyone
    
    Best regards,
    The R-MAN E-Commerce Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
  `;

  await sendEmail({ to: email, subject, html, text });
};

/**
 * Send password changed confirmation email
 * @param {String} email - Recipient email
 * @param {String} fullName - User's full name
 */
export const sendPasswordChangedEmail = async (email, fullName) => {
  const subject = 'Password Changed Successfully - R-MAN E-Commerce';
  
  const html = `
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
          .container { max-width: 600px; margin: 0 auto; padding: 20px; }
          .header { background-color: #4CAF50; color: white; padding: 20px; text-align: center; }
          .content { padding: 20px; background-color: #f9f9f9; }
          .footer { padding: 20px; text-align: center; font-size: 12px; color: #666; }
          .success { background-color: #d4edda; border-left: 4px solid #28a745; padding: 10px; margin: 15px 0; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Password Changed Successfully</h1>
          </div>
          <div class="content">
            <h2>Hello ${fullName},</h2>
            <div class="success">
              <strong>✓ Your password has been changed successfully!</strong>
            </div>
            <p>This is a confirmation that your R-MAN E-Commerce account password was recently changed.</p>
            <p><strong>Changed on:</strong> ${new Date().toLocaleString()}</p>
            <p>If you did not make this change, please contact our support team immediately.</p>
            <p>Best regards,<br>The R-MAN E-Commerce Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
  `;

  const text = `
    Password Changed Successfully - R-MAN E-Commerce
    
    Hello ${fullName},
    
    ✓ Your password has been changed successfully!
    
    This is a confirmation that your R-MAN E-Commerce account password was recently changed.
    
    Changed on: ${new Date().toLocaleString()}
    
    If you did not make this change, please contact our support team immediately.
    
    Best regards,
    The R-MAN E-Commerce Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
  `;

  await sendEmail({ to: email, subject, html, text });
};

/**
 * Send email verification email
 * @param {String} email - Recipient email
 * @param {String} fullName - User's full name
 * @param {String} verificationToken - Email verification token
 */
export const sendEmailVerificationEmail = async (email, fullName, verificationToken) => {
  const verificationUrl = `${process.env.EMAIL_VERIFICATION_URL || 'http://localhost:3000/verify-email'}?token=${verificationToken}`;
  const subject = 'Verify Your Email - R-MAN E-Commerce';
  
  const html = `
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
          .container { max-width: 600px; margin: 0 auto; padding: 20px; }
          .header { background-color: #2196F3; color: white; padding: 20px; text-align: center; }
          .content { padding: 20px; background-color: #f9f9f9; }
          .footer { padding: 20px; text-align: center; font-size: 12px; color: #666; }
          .button { display: inline-block; padding: 12px 30px; background-color: #2196F3; color: white; text-decoration: none; border-radius: 5px; margin: 15px 0; }
          .info { background-color: #e3f2fd; border-left: 4px solid #2196F3; padding: 10px; margin: 15px 0; }
          .token-box { background-color: #e9ecef; padding: 15px; border-radius: 5px; font-family: monospace; word-break: break-all; margin: 15px 0; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Verify Your Email Address</h1>
          </div>
          <div class="content">
            <h2>Hello ${fullName},</h2>
            <p>Thank you for registering with R-MAN E-Commerce!</p>
            <p>Please verify your email address by clicking the button below:</p>
            <div style="text-align: center;">
              <a href="${verificationUrl}" class="button">Verify Email Address</a>
            </div>
            <p>Or copy and paste this link into your browser:</p>
            <div class="token-box">${verificationUrl}</div>
            <div class="info">
              <strong>ℹ️ Note:</strong>
              <ul style="margin: 5px 0;">
                <li>This link will expire in 24 hours</li>
                <li>If you didn't create an account, please ignore this email</li>
              </ul>
            </div>
            <p>Best regards,<br>The R-MAN E-Commerce Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
  `;

  const text = `
    Verify Your Email Address - R-MAN E-Commerce
    
    Hello ${fullName},
    
    Thank you for registering with R-MAN E-Commerce!
    
    Please verify your email address by clicking the link below:
    ${verificationUrl}
    
    ℹ️ Note:
    - This link will expire in 24 hours
    - If you didn't create an account, please ignore this email
    
    Best regards,
    The R-MAN E-Commerce Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
  `;

  await sendEmail({ to: email, subject, html, text });
};

export default {
  sendWelcomeEmail,
  sendPasswordResetEmail,
  sendPasswordChangedEmail,
  sendEmailVerificationEmail,
};

File: back-end/cmps/app/services/email_service.py
"""
Email Service
Handles sending complaint-related email notifications
"""
from app.config.email import send_email
import logging

logger = logging.getLogger(__name__)


def send_complaint_created_email(customer_email: str, customer_name: str, complaint_id: str, subject: str, category: str):
    """
    Send complaint registration confirmation email to customer
    
    Args:
        customer_email: Customer email address
        customer_name: Customer full name
        complaint_id: Complaint ID
        subject: Complaint subject
        category: Complaint category
    """
    email_subject = f"Complaint Registered: {complaint_id} - R-MAN E-Commerce"
    
    html_body = f"""
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
          .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
          .header {{ background-color: #FF9800; color: white; padding: 20px; text-align: center; }}
          .content {{ padding: 20px; background-color: #f9f9f9; }}
          .complaint-box {{ background-color: #fff; border-left: 4px solid #FF9800; padding: 15px; margin: 15px 0; }}
          .footer {{ padding: 20px; text-align: center; font-size: 12px; color: #666; }}
          .info {{ background-color: #e3f2fd; padding: 10px; margin: 15px 0; border-radius: 4px; }}
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Complaint Registered Successfully</h1>
          </div>
          <div class="content">
            <h2>Hello {customer_name},</h2>
            <p>Thank you for reaching out to us. We have successfully registered your complaint and our team will review it shortly.</p>
            
            <div class="complaint-box">
              <p><strong>Complaint ID:</strong> {complaint_id}</p>
              <p><strong>Category:</strong> {category}</p>
              <p><strong>Subject:</strong> {subject}</p>
              <p><strong>Status:</strong> Open</p>
            </div>
            
            <div class="info">
              <strong>ℹ️ What happens next?</strong>
              <ul style="margin: 5px 0;">
                <li>Our team will review your complaint within 24 hours</li>
                <li>You'll receive updates via email when status changes</li>
                <li>You can add comments to provide more information</li>
                <li>Track your complaint status anytime through your account</li>
              </ul>
            </div>
            
            <p>You can reference this complaint using ID: <strong>{complaint_id}</strong></p>
            
            <p>We appreciate your patience and will work to resolve your concern as quickly as possible.</p>
            
            <p>Best regards,<br>The R-MAN Customer Support Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
            <p>If you have any questions, please contact our support team.</p>
          </div>
        </div>
      </body>
    </html>
    """
    
    text_body = f"""
    Complaint Registered Successfully - R-MAN E-Commerce
    
    Hello {customer_name},
    
    Thank you for reaching out to us. We have successfully registered your complaint.
    
    Complaint Details:
    - Complaint ID: {complaint_id}
    - Category: {category}
    - Subject: {subject}
    - Status: Open
    
    What happens next?
    - Our team will review your complaint within 24 hours
    - You'll receive updates via email when status changes
    - You can add comments to provide more information
    - Track your complaint status anytime through your account
    
    Reference: {complaint_id}
    
    Best regards,
    The R-MAN Customer Support Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
    """
    
    send_email(customer_email, email_subject, html_body, text_body)


def send_complaint_status_changed_email(customer_email: str, customer_name: str, complaint_id: str, 
                                       old_status: str, new_status: str, subject: str):
    """
    Send email notification when complaint status changes
    
    Args:
        customer_email: Customer email address
        customer_name: Customer full name
        complaint_id: Complaint ID
        old_status: Previous status
        new_status: New status
        subject: Complaint subject
    """
    email_subject = f"Status Update: {complaint_id} - R-MAN E-Commerce"
    
    # Status-specific messages
    status_messages = {
        "In Progress": "Our team is actively working on your complaint.",
        "Resolved": "Your complaint has been resolved! Please review the resolution details.",
        "Closed": "Your complaint has been closed. Thank you for your patience.",
        "Reopened": "Your complaint has been reopened and will be reviewed again.",
    }
    
    status_message = status_messages.get(new_status, "Your complaint status has been updated.")
    
    html_body = f"""
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
          .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
          .header {{ background-color: #2196F3; color: white; padding: 20px; text-align: center; }}
          .content {{ padding: 20px; background-color: #f9f9f9; }}
          .status-box {{ background-color: #fff; border-left: 4px solid #2196F3; padding: 15px; margin: 15px 0; }}
          .footer {{ padding: 20px; text-align: center; font-size: 12px; color: #666; }}
          .highlight {{ background-color: #e8f5e9; padding: 10px; margin: 15px 0; border-radius: 4px; }}
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Complaint Status Update</h1>
          </div>
          <div class="content">
            <h2>Hello {customer_name},</h2>
            <p>{status_message}</p>
            
            <div class="status-box">
              <p><strong>Complaint ID:</strong> {complaint_id}</p>
              <p><strong>Subject:</strong> {subject}</p>
              <p><strong>Previous Status:</strong> {old_status}</p>
              <p><strong>New Status:</strong> <span style="color: #2196F3; font-weight: bold;">{new_status}</span></p>
            </div>
            
            <p>You can view complete details and add comments by logging into your account.</p>
            
            <p>Best regards,<br>The R-MAN Customer Support Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
    """
    
    text_body = f"""
    Complaint Status Update - R-MAN E-Commerce
    
    Hello {customer_name},
    
    {status_message}
    
    Complaint Details:
    - Complaint ID: {complaint_id}
    - Subject: {subject}
    - Previous Status: {old_status}
    - New Status: {new_status}
    
    You can view complete details and add comments by logging into your account.
    
    Best regards,
    The R-MAN Customer Support Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
    """
    
    send_email(customer_email, email_subject, html_body, text_body)


def send_complaint_resolved_email(customer_email: str, customer_name: str, complaint_id: str, 
                                  subject: str, resolution_notes: str, resolved_by_name: str):
    """
    Send email notification when complaint is resolved
    
    Args:
        customer_email: Customer email address
        customer_name: Customer full name
        complaint_id: Complaint ID
        subject: Complaint subject
        resolution_notes: Resolution details
        resolved_by_name: Name of admin who resolved
    """
    email_subject = f"Complaint Resolved: {complaint_id} - R-MAN E-Commerce"
    
    html_body = f"""
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
          .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
          .header {{ background-color: #4CAF50; color: white; padding: 20px; text-align: center; }}
          .content {{ padding: 20px; background-color: #f9f9f9; }}
          .resolution-box {{ background-color: #fff; border-left: 4px solid #4CAF50; padding: 15px; margin: 15px 0; }}
          .footer {{ padding: 20px; text-align: center; font-size: 12px; color: #666; }}
          .success {{ background-color: #e8f5e9; padding: 10px; margin: 15px 0; border-radius: 4px; }}
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>✓ Complaint Resolved</h1>
          </div>
          <div class="content">
            <h2>Hello {customer_name},</h2>
            
            <div class="success">
              <p><strong>Good news!</strong> Your complaint has been resolved.</p>
            </div>
            
            <div class="resolution-box">
              <p><strong>Complaint ID:</strong> {complaint_id}</p>
              <p><strong>Subject:</strong> {subject}</p>
              <p><strong>Resolved By:</strong> {resolved_by_name}</p>
              <p><strong>Resolution:</strong></p>
              <p style="white-space: pre-wrap;">{resolution_notes}</p>
            </div>
            
            <p><strong>Not satisfied with the resolution?</strong></p>
            <p>If you're not satisfied with the resolution, you can reopen this complaint within a reasonable timeframe by logging into your account.</p>
            
            <p>Thank you for your patience and for giving us the opportunity to resolve your concern.</p>
            
            <p>Best regards,<br>The R-MAN Customer Support Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
    """
    
    text_body = f"""
    Complaint Resolved - R-MAN E-Commerce
    
    Hello {customer_name},
    
    Good news! Your complaint has been resolved.
    
    Complaint Details:
    - Complaint ID: {complaint_id}
    - Subject: {subject}
    - Resolved By: {resolved_by_name}
    
    Resolution:
    {resolution_notes}
    
    Not satisfied with the resolution?
    If you're not satisfied with the resolution, you can reopen this complaint by logging into your account.
    
    Thank you for your patience and for giving us the opportunity to resolve your concern.
    
    Best regards,
    The R-MAN Customer Support Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
    """
    
    send_email(customer_email, email_subject, html_body, text_body)


def send_complaint_assigned_email(customer_email: str, customer_name: str, complaint_id: str, 
                                 subject: str, assigned_to_name: str):
    """
    Send email notification when complaint is assigned to an admin
    
    Args:
        customer_email: Customer email address
        customer_name: Customer full name
        complaint_id: Complaint ID
        subject: Complaint subject
        assigned_to_name: Name of admin assigned
    """
    email_subject = f"Complaint Update: {complaint_id} - R-MAN E-Commerce"
    
    html_body = f"""
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
          .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
          .header {{ background-color: #9C27B0; color: white; padding: 20px; text-align: center; }}
          .content {{ padding: 20px; background-color: #f9f9f9; }}
          .info-box {{ background-color: #fff; border-left: 4px solid #9C27B0; padding: 15px; margin: 15px 0; }}
          .footer {{ padding: 20px; text-align: center; font-size: 12px; color: #666; }}
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Complaint Assigned</h1>
          </div>
          <div class="content">
            <h2>Hello {customer_name},</h2>
            <p>Your complaint has been assigned to a specialist for review.</p>
            
            <div class="info-box">
              <p><strong>Complaint ID:</strong> {complaint_id}</p>
              <p><strong>Subject:</strong> {subject}</p>
              <p><strong>Assigned To:</strong> {assigned_to_name}</p>
              <p><strong>Status:</strong> In Progress</p>
            </div>
            
            <p>Our specialist will review your complaint and work towards a resolution. You'll receive updates as progress is made.</p>
            
            <p>Best regards,<br>The R-MAN Customer Support Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
    """
    
    text_body = f"""
    Complaint Assigned - R-MAN E-Commerce
    
    Hello {customer_name},
    
    Your complaint has been assigned to a specialist for review.
    
    Complaint Details:
    - Complaint ID: {complaint_id}
    - Subject: {subject}
    - Assigned To: {assigned_to_name}
    - Status: In Progress
    
    Our specialist will review your complaint and work towards a resolution.
    
    Best regards,
    The R-MAN Customer Support Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
    """
    
    send_email(customer_email, email_subject, html_body, text_body)


def send_complaint_comment_email(customer_email: str, customer_name: str, complaint_id: str, 
                                subject: str, commenter_name: str, comment: str):
    """
    Send email notification when a new comment is added to complaint
    
    Args:
        customer_email: Customer email address
        customer_name: Customer full name
        complaint_id: Complaint ID
        subject: Complaint subject
        commenter_name: Name of person who added comment
        comment: Comment text
    """
    email_subject = f"New Comment: {complaint_id} - R-MAN E-Commerce"
    
    html_body = f"""
    <!DOCTYPE html>
    <html>
      <head>
        <style>
          body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
          .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
          .header {{ background-color: #00BCD4; color: white; padding: 20px; text-align: center; }}
          .content {{ padding: 20px; background-color: #f9f9f9; }}
          .comment-box {{ background-color: #fff; border-left: 4px solid #00BCD4; padding: 15px; margin: 15px 0; }}
          .footer {{ padding: 20px; text-align: center; font-size: 12px; color: #666; }}
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>New Comment Added</h1>
          </div>
          <div class="content">
            <h2>Hello {customer_name},</h2>
            <p>A new comment has been added to your complaint.</p>
            
            <div class="comment-box">
              <p><strong>Complaint ID:</strong> {complaint_id}</p>
              <p><strong>Subject:</strong> {subject}</p>
              <p><strong>Comment By:</strong> {commenter_name}</p>
              <p><strong>Comment:</strong></p>
              <p style="white-space: pre-wrap;">{comment}</p>
            </div>
            
            <p>You can view all comments and reply by logging into your account.</p>
            
            <p>Best regards,<br>The R-MAN Customer Support Team</p>
          </div>
          <div class="footer">
            <p>&copy; 2026 R-MAN Corporation, Bangalore. All rights reserved.</p>
          </div>
        </div>
      </body>
    </html>
    """
    
    text_body = f"""
    New Comment Added - R-MAN E-Commerce
    
    Hello {customer_name},
    
    A new comment has been added to your complaint.
    
    Complaint Details:
    - Complaint ID: {complaint_id}
    - Subject: {subject}
    - Comment By: {commenter_name}
    
    Comment:
    {comment}
    
    You can view all comments and reply by logging into your account.
    
    Best regards,
    The R-MAN Customer Support Team
    
    © 2026 R-MAN Corporation, Bangalore. All rights reserved.
    """
    
    send_email(customer_email, email_subject, html_body, text_body)
